Library/Homebrew/extend/ARGV.rb

```diff
@@ -254,7 +254,7 @@ def spec(default = :stable)
   def downcased_unique_named
     # Only lowercase names, not paths, bottle filenames or URLs
     @downcased_unique_named ||= named.map do |arg|
-      if arg.include?("/") || arg.end_with?(".tar.gz")
+      if arg.include?("/") || arg.end_with?(".tar.gz") || File.exist?(arg)
```

Comment:
Similar comment here about a bit stricter checking. I don't think it's helpful to accept arbitrary file names that might or might not exist in the current working directory. Maybe it's worth doing the check in a separate method named `formula_file?` as this check is already done in two different locations.

From the comment above this: “not paths, bottle filenames”. – This doesn't seem to be the case, and apparently already for a long time. 😕